### PR TITLE
AdvanceIfNeeded on bitmap container optimized

### PR DESCRIPTION
**Pull request by xtonik**
### SUMMARY
Action `advanceIfNeeded()` on bitmap container was not optimal as noted in source code - too complicated bit-wise iterating within one long value, where given limit value belongs to. Above that, when this long value (64-bit bitmap) is not zero and no value under / above limit was found, iterating is done by bits, not by bitmap longs as probably supposed by author of code.

Temporary performance comparison can be seen by running `AdvanceIfNeededBenchmark` when 48ac7f9 is checked out. According to my tests, 10% performance improvement reached.

Tiny related code cleanup done also.

### Automated Checks

- [x] I have run `./gradlew test` and made sure that my PR does not break any unit test.
- [x] I have run `./gradlew checkstyleMain` or the equivalent and corrected the formatting warnings reported.

### CODE REVIEW

**Comment by lemire** [comment]
@xtonik This is good. Can you check my comment regarding the benchmark and update the code so that you avoid `Math.abs` which seems unnecessary.

File: jmh/src/jmh/java/org/roaringbitmap/AdvanceIfNeededBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+public class AdvanceIfNeededBenchmark {
+    public static MutableRoaringBitmap c2;
+    public static PeekableIntIterator it;
+
+    private static int[] takeSortedAndDistinct(Random source, int count) {
+        LinkedHashSet<Integer> ints = new LinkedHashSet<>(count);
+        for (int size = 0; size < count; size++) {
+            int next;
+            do {
+                next = Math.abs(source.nextInt());
```
[the remark this answers is not in the document]
Comment (xtonik) [reply jmh/src/jmh/java/org/roaringbitmap/AdvanceIfNeededBenchmark.java:26]: So what you suggest concretely to be clear?

Comment (lemire) [inline jmh/src/jmh/java/org/roaringbitmap/AdvanceIfNeededBenchmark.java:26]:
```suggestion
                next = source.nextInt(1000000);
```

[the remark this answers is not in the document]
Comment (lemire) [reply jmh/src/jmh/java/org/roaringbitmap/AdvanceIfNeededBenchmark.java:26]: Sure. See my proposal.

**Comment by lemire** [comment]
Good. Merging.

**Pull request merged**